### PR TITLE
#159993338 Login via Google and Facebook

### PR DESCRIPTION
**Pull request by hoslack**
#### What does this PR do?
- Enables user to login with Facebook and Google

#### Description of Task to be completed?
- Create facebook and google login buttons
- Get the tokens from the auth providers above
- Send the auth token from the providers to the backend API for verification using a redux action
- Update the reducer with user details after getting verification from the backend 
- Update the localstorage with current user details
- Redirect the user to the home page

#### How should this be manually tested?
- Git pull the source code from ft-Login-via-Google-Facebook-Twitter-159993338 branch
- Copy the variables from sample.env and put the correct parameters in .env file
- Run `npm i` to install the dependencies used in the project
- Run `npm start` to start the project
- Go to the browser and try loging in with Facebook and Google
#### What are the relevant pivotal tracker stories?
[#159993338](https://www.pivotaltracker.com/n/projects/2184741/stories/159993338)
#### Screenshots (if appropriate)

- Login Page
screen shot 2018-09-05 at 00 57 06

- During Login
screen shot 2018-09-05 at 00 57 26

- After Login
screen shot 2018-09-05 at 00 57 46

### CODE REVIEW

File: src/actions/types.js [hunk 1]
```diff
@@ -1 +1,4 @@
-export const TEST_REDUX = 'TEST_REDUX';
+export const BEGIN_LOGIN = 'BEGIN_LOGIN';
```
[the remark this answers is not in the document]
Comment (RuthNjeri) [reply src/actions/types.js]: Add `.eslintrc.json`  in the Hound yml file for it to detect the eslint configurations.

[the remark this answers is not in the document]
Comment (hoslack) [reply src/actions/types.js]: I have just done that @RuthNjeri . Let me see the new results

File: src/reducers/loginReducer.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import * as types from '../actions/types';
```
[the remark this answers is not in the document]
Comment (davenjeru) [reply src/reducers/loginReducer.js:1]: Added "babel-eslint" as parser and editted the "parserOptions" to allow "sorceType:  module"

File: src/index.js [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+  {},
+  composeWithDevTools(applyMiddleware(Thunk)),
+);
 
 ReactDOM.render(
   <Provider store={store}>
     <App />
   </Provider>,
-  document.getElementById("root")
+  document.getElementById('root'),
```
[the remark this answers is not in the document]
Comment (davenjeru) [reply src/index.js]: Added "browser: true" in "env:{}" to allow the use of the 'document' keyword.

File: src/components/LoginForm.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    const errors = this.validate(this.state.data);
+    this.setState({ errors });
+    if (Object.keys(errors).length === 0) {
+      this.setState({ loading: true });
+      this.props
+        .submit(this.state.data)
+        .catch(err => this.setState({ errors: err.response.data.errors, loading: false }));
```
[the remark this answers is not in the document]
Comment (davenjeru) [reply src/components/LoginForm.js]: @blac-siren should take care of LoginForm.js

**Pull request merged**